packages/react/src/components/Notification/Notification.tsx

```diff
@@ -567,17 +805,118 @@ InlineNotification.propTypes = {
    */
   title: PropTypes.string,
 };
-
 InlineNotification.defaultProps = {
   kind: 'error',
   role: 'status',
   onCloseButtonClick: () => {},
   hideCloseButton: false,
 };
 
+/**
+ * ActionableNotification
+ * ======================
+ */
+
+export interface ActionableNotificationProps
+  extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * Pass in the action button label that will be rendered within the ActionableNotification.
+   */
+  actionButtonLabel: string;
+
+  /**
+   * Provide a description for "close" icon button that can be read by screen readers
+   */
+  'aria-label'?: string;
+
+  /**
+   * Specify the content
+   */
+  children?: ReactNode;
+
+  /**
+   * Specify an optional className to be applied to the notification box
+   */
+  className?: string;
+
+  /**
+   * Specify if pressing the escape key should close notifications
+   */
+  closeOnEscape?: boolean;
+
+  /**
+   * Specify if focus should be moved to the component when the notification contains actions
+   */
+  hasFocus?: boolean;
+
+  /**
+   * Specify the close button should be disabled, or not
+   */
+  hideCloseButton?: boolean;
+
+  /*
+   * Specify if the notification should have inline styling applied instead of toast
+   */
+  inline?: boolean;
+
+  /**
+   * Specify what state the notification represents
+   */
+  kind:
```

Comment:
I'm thinking this could be refactored to follow our pattern in https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Button/Button.tsx

@francinelucca thoughts? Might not be needed for this PR

Reply:
I'll push up separately since it's not needed for this PR